les/peer.go

```diff
@@ -1012,6 +1011,9 @@ func (p *clientPeer) Handshake(td *big.Int, head common.Hash, headNum uint64, ge
 	if recentTx != txIndexUnlimited && p.version < lpv4 {
 		return errors.New("Cannot serve old clients without a complete tx index")
 	}
+	if server.recentState <= BlockSafetyMargin {
```

Comment:
If we can ensure this value is always larger than `BlockSafetyMargin`, then we don't need this check.